src/common/string_utils.hpp

```diff
@@ -0,0 +1,29 @@
+/* Copyright (C) 2021, Nikolai Wuttke. All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace rigel::strings {
```

Comment:
I'd prefer to have this file in `src/base` instead of `src/common`. The former is where I mostly have general purpose utilities/standard library extensions, and string utilities fit in there quite well.

`common` on the other hand is more of a place for things that are needed to glue `frontend` and the rest of the code together - at least that's the idea, but in practice it mostly ended up being the place where I put things that don't really fit anywhere else 😅
 
Admittedly, that's not very clear from the name of the directory, I should think about a better name at some point..